kraken/pvc/pvc_scenario.py

```diff
@@ -46,65 +46,52 @@ def run(scenarios_list, config):
-                    if len(entry["persistentVolumeClaim"]["claimName"]) > 0:
-                        volume_name = entry["name"]
-                        pvc_name = entry["persistentVolumeClaim"]["claimName"]
+                pod = kubecli.get_pod_info(name=pod_name, namespace=namespace)
+                
+                if pod is None:
+                    logging.error("Exiting as pod '%s' doesn't exist in namespace '%s'" % (str(pod_name), str(namespace)))
+                    sys.exit(1)
+                
+                for volume in pod.volumes:
+                    if volume.pvcName is not None:
+                        volume_name = volume.name
+                        pvc_name = volume.pvcName
+                        pvc = kubecli.get_pvc_info(pvc_name, namespace)
                         break
+                if 'pvc' not in locals():
+                    logging.error(
+                        "Pod '%s' in namespace '%s' does not use a pvc" % (str(pod_name), str(namespace))
+                    )
+                    sys.exit(1)
                 logging.info("Volume name: %s" % volume_name)
                 logging.info("PVC name: %s" % pvc_name)
 
                 # Get container name and mount path
-                command = 'kubectl get pods %s -n %s -o json | jq -r ".spec.containers"' % (
-                    str(pod_name),
-                    str(namespace),
-                )
-                logging.debug("Get mount path command:\n %s" % command)
-                volume_mounts_list = runcommand.invoke(command, 60).rstrip().replace("\n]\n[\n", ",\n")
-                volume_mounts_list_json = json.loads(volume_mounts_list)
-                for entry in volume_mounts_list_json:
-                    for vol in entry["volumeMounts"]:
-                        if vol["name"] == volume_name:
-                            mount_path = vol["mountPath"]
-                            container_name = entry["name"]
+                for container in  pod.containers:
+                    for vol in container.volumeMounts:
+                        if vol.name == volume_name:
+                            mount_path = vol.mountPath
+                            container_name = container.name
                             break
                 logging.info("Container path: %s" % container_name)
                 logging.info("Mount path: %s" % mount_path)
 
                 # Get PVC capacity
-                command = "kubectl describe pvc %s -n %s | grep \"Capacity:\" | grep -Eo '[^: ]*$'" % (
-                    str(pvc_name),
-                    str(namespace),
-                )
-                pvc_capacity = runcommand.invoke(
-                    command,
-                    60,
-                ).rstrip()
-                logging.debug("Get PVC capacity command:\n %s" % command)
-                pvc_capacity_kb = toKbytes(pvc_capacity)
+                command = "df %s -B 1024 | sed 1d | awk -F' ' '{print $4}'" % (str(mount_path))
```

Comment:
thoughts on combining this line with line 98 to only have to do the exec call once, then do the parsing of the output in the script to get the capacity and used bytes

Reply:
That makes sense, I've modified it now